test/iidm/SubstationTest.cpp

```diff
@@ -443,6 +446,63 @@ BOOST_AUTO_TEST_CASE(threeWindingsTransformerCount) {
     BOOST_CHECK_EQUAL(5UL, substation.getThreeWindingsTransformerCount());
 }
 
+BOOST_AUTO_TEST_CASE(remove) {
+    Network network = powsybl::network::FourSubstationsNodeBreakerFactory::create();
+    Substation& s1 = network.getSubstation("S1");
+    Substation& s2 = network.getSubstation("S2");
+    Substation& s3 = network.getSubstation("S3");
+    Substation& s4 = network.getSubstation("S4");
+
+    POWSYBL_ASSERT_THROW(s1.remove(), PowsyblException, "The substation S1 is still connected to another substation");
+    POWSYBL_ASSERT_THROW(s2.remove(), PowsyblException, "The substation S2 is still connected to another substation");
+    POWSYBL_ASSERT_THROW(s3.remove(), PowsyblException, "The substation S3 is still connected to another substation");
+    POWSYBL_ASSERT_THROW(s4.remove(), PowsyblException, "The substation S4 is still connected to another substation");
+
+    VoltageLevel& s1VL1 = network.getVoltageLevel("S1VL1");
+
+    POWSYBL_ASSERT_THROW(s1VL1.remove(), AssertionError, "The voltage level 'S1VL1' cannot be removed because of a remaining TWO_WINDINGS_TRANSFORMER");
+
+    POWSYBL_ASSERT_THROW(s1VL1.remove(), AssertionError, "The voltage level 'S1VL1' cannot be removed because of a remaining TWO_WINDINGS_TRANSFORMER");
+    network.getTwoWindingsTransformer("TWT").remove();
+
+    s1VL1.remove();
+    POWSYBL_ASSERT_THROW(s1.remove(), PowsyblException, "The substation S1 is still connected to another substation");
+
+    VoltageLevel& s2vl1 = network.getVoltageLevel("S2VL1");
+    POWSYBL_ASSERT_THROW(s2vl1.remove(), AssertionError, "The voltage level 'S2VL1' cannot be removed because of a remaining HVDC line");
+
+    VoltageLevel& s3VL1 = network.getVoltageLevel("S3VL1");
+    POWSYBL_ASSERT_THROW(s3VL1.remove(), AssertionError, "The voltage level 'S3VL1' cannot be removed because of a remaining LINE");
+
+    VoltageLevel& s4VL1 = network.getVoltageLevel("S4VL1");
+    POWSYBL_ASSERT_THROW(s4VL1.remove(), AssertionError, "The voltage level 'S4VL1' cannot be removed because of a remaining LINE");
+
+    // s3 and s4 are linked through LINE_S3S4
+    network.getLine("LINE_S3S4").remove();
+
+    // s4 is removable
+    s4.remove();
+
+    // s2 and s3 are linked by Line LINE_S2S3
+    network.getLine("LINE_S2S3").remove();
+
+    // s3 linked to s1 thought HVDC2
+    network.getHvdcLine("HVDC2").remove();
+    network.getVscConverterStation("VSC2").remove();
+    network.getLccConverterStation("LCC2").remove();
+
+    // s3 is isolated => removable
+    s3.remove();
+
+    // s1 and s2 are linked by HVDC1
+    network.getVscConverterStation("VSC1").remove();
+    network.getLccConverterStation("LCC1").remove();
```

Comment:
To be fixed: the HVDC1 line should be removed before being able to remove its converter station